agent/agent.go

```diff
@@ -285,6 +285,9 @@ func (a *Agent) run(ctx context.Context) {
 				if err := a.worker.Assign(ctx, msg.Changes); err != nil {
 					log.G(ctx).WithError(err).Error("failed to synchronize worker assignments")
 				}
+				// then, call Listen on the worker again to update all of the
+				// task statuses.
+				a.worker.Listen(ctx, reporter)
```

Comment:
I suggest adding a function which gets all the updates instead of calling Listen() again, which is confusing to read.

Reply:
Yeah, will do this.